GW2EIEvtcParser/EIData/InstantCastFinders/CastFinderHelpers.cs

```diff
@@ -50,6 +50,12 @@ internal static bool HasGainedBuff(CombatData combatData, long buffID, AgentItem
                 .Any(apply => apply.To == agent && apply.CreditedBy == source && Math.Abs(apply.AppliedDuration - appliedDuration) < epsilon);
         }
 
+        internal static bool HasLostBuff(CombatData combatData, long buffID, AgentItem agent, long time, long epsilon = ServerDelayConstant)
+        {
+            return FindRelatedEvents(combatData.GetBuffData(buffID).OfType<AbstractBuffRemoveEvent>(), time, epsilon)
```

Comment:
That would also trigger on BuffRemoveSingle events, is that the intention behind the method, as the buff wouldn't be completely lost but only one stack removed? Otherwise would be better to limit to BuffRemoveAll events. We can add a method to specifically track one stack getting removed later if needed.

Reply:
It was intentional since I don't know how many stacks get removed by Swap when you would somehow manage to get more than 1 stack of the Illusionary Leap buff. No idea if there is even a way to achieve that. I think it makes sense to rename this to something like `HasLostBuffStack` and add another one for losing last/all.